### PR TITLE
kak-lsp: init at 6.2.1

### PR DESCRIPTION
**Pull request by spacekookie**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
@grahamcofborg eval

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build kak-lsp

**Comment by worldofpeace** [comment]
Hi @spacekookie. This expression looks good, could you give some pointers on how I should test its function?

**Comment by risicle** [comment]
To get this building on darwin you just have to add

```
  buildInputs = lib.optional stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
```

obviously having added `darwin` and `stdenv` to your args...

**Comment by spacekookie** [comment]
> Hi @spacekookie. This expression looks good, could you give some pointers on how I should test its function?

It's functionality is pretty centered around kakoune. What this
requires is a config file, that can be found in the project repo [1],
and something to hook into it from your kakoune [2].

[1]: https://github.com/ul/kak-lsp/blob/master/kak-lsp.toml
[2]: https://github.com/ul/kak-lsp#usage

In the future, it might be interesting to build a module around
configuring the LSP or hooking it into the kakoune plugin system in
nixpkgs (that I haven't played around with yet). But just having the
software itself packaged is gonna be helpful to a bunch of people who
want to use it already.

**Comment by spacekookie** [comment]
@risicle Ah nice, thank you for that tip! I just adjusted the derivation

File: pkgs/tools/misc/kak-lsp/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, fetchFromGitHub, rustPlatform }:
```
Comment (worldofpeace) [inline pkgs/tools/misc/kak-lsp/default.nix]:
```suggestion
{stdenv, lib, fetchFromGitHub, rustPlatform }:
```

fix eval

Comment (spacekookie) [reply pkgs/tools/misc/kak-lsp/default.nix]: Ah, thanks! Completely missed that :sweat_smile:

Comment (worldofpeace) [reply pkgs/tools/misc/kak-lsp/default.nix]: oops we need `darwin` too

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build kak-lsp

**Comment by worldofpeace** [comment]
Thanks @spacekookie

**Pull request merged**